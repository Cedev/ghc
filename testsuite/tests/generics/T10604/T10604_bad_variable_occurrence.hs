{-# LANGUAGE DeriveGeneric #-}
{-# LANGUAGE PolyKinds #-}
module T10604_bad_variable_occurrence where

import GHC.Generics

newtype Twice f = Twice ((f Int) Int)
  deriving Generic1
