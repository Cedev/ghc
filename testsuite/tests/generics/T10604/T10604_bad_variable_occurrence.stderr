
T10604_bad_variable_occurrence.hs:8:12: error:
    • Can't make a derived instance of ‘Generic1 Twice’:
        Constructor ‘Twice’ applies a type to an argument involving the last parameter
                         but the applied type is not of kind * -> *
    • In the newtype declaration for ‘Twice’
